### DISCUSSION TITLE
Can't get data from tables with RLS for authenticated users

### DISCUSSION THREAD

**Discussion by rezirezi** [supabase/supabase#15101] [Questions] [answered]
I have pinpointed the problem, which is the fact that I need to include the access token of the users when getting data, however, I just can't figure out how, and all the similar discussions I found and their solutions are either outdated or don't work. I am using next js to build the app. basically copied this https://github.com/mryechkin/nextjs-supabase-auth

**Comment by tholder** [comment, marked as answer]
Your question isn't well formed. However, assuming you are talking about RLS and returning data base on auth'd user, your RLS rules should use auth.uid() function somewhere in them. Either to check a row is owned by this user or perhaps they have access through another table in which case see my answer here https://github.com/orgs/supabase/discussions/15087

**Reply by rezirezi** [reply]
thanks!! turns out my policies were wrong, for some reason i thought i tested them before and they worked
